homeassistant/components/group/sensor.py

```diff
@@ -0,0 +1,411 @@
+        """Register callbacks."""
+
+        @callback
+        def async_state_changed_listener(event: Event) -> None:
+            """Handle child updates."""
+            self.async_set_context(event.context)
+            self.async_defer_or_update_ha_state()
+
+        self.async_on_remove(
+            async_track_state_change_event(
+                self.hass, self._entity_ids, async_state_changed_listener
+            )
+        )
+
+        await super().async_added_to_hass()
+
+    @callback
+    def async_update_group_state(self) -> None:
+        """Query all members and determine the sensor group state."""
+        states: list[StateType] = []
+        valid_states: list[bool] = []
+        sensor_values: list[tuple[str, float, State]] = []
+        for entity_id in self._entity_ids:
+            if (state := self.hass.states.get(entity_id)) is not None:
+                states.append(state.state)
+                try:
+                    sensor_values.append((entity_id, float(state.state), state))
+                    if entity_id in self._state_incorrect:
+                        self._state_incorrect.remove(entity_id)
+                except ValueError:
+                    valid_states.append(False)
+                    if entity_id not in self._state_incorrect:
+                        self._state_incorrect.add(entity_id)
+                        _LOGGER.warning(
+                            "Unable to use state. Only numerical states are supported, entity %s with value %s excluded from calculation",
+                            entity_id,
+                            state.state,
+                        )
+                    continue
+                valid_states.append(True)
+
+        # Set group as unavailable if any/all members do not have numeric values
+        self._attr_available = any(state != STATE_UNAVAILABLE for state in states)
+
+        valid_state = self.mode(
+            state not in (STATE_UNKNOWN, STATE_UNAVAILABLE) for state in states
+        )
+        valid_state_numeric = self.mode(numeric_state for numeric_state in valid_states)
+
+        if not valid_state or not valid_state_numeric:
+            self._attr_native_value = None
+            return
+
+        # Calculate values
+        self._calculate_entity_properties()
+        self._extra_state_attribute, self._attr_native_value = self._state_calc(
+            sensor_values, self._round_digits
+        )
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
+        """Return the state attributes of the sensor."""
+        return {ATTR_ENTITY_ID: self._entity_ids, **self._extra_state_attribute}
+
+    @property
+    def device_class(self) -> SensorDeviceClass | None:
+        """Return device class."""
+        if self._attr_device_class is not None:
+            return self._attr_device_class
+        return self.calc_device_class
+
+    @property
+    def state_class(self) -> SensorStateClass | str | None:
```

Comment:
It doesn't seem possible to return a `str` here
```suggestion
    def state_class(self) -> SensorStateClass | None:
```